BLE_GAP/source/main.cpp

```diff
@@ -499,19 +515,13 @@ class GapDemo : private mbed::NonCopyable<GapDemo>, public ble::Gap::EventHandle
                     print_error(error, "Error caused by Gap::stopAdvertising");
                 }
             }
-
-            ble_error_t error = _gap.destroyAdvertisingSet(_extended_adv_handle);
-            if (error) {
-                print_error(error, "Error caused by Gap::destroyAdvertisingSet");
-            }
-
-            _extended_adv_handle = ble::INVALID_ADVERTISING_HANDLE;
         }
-#endif // BLE_FEATURE_EXTENDED_ADVERTISING
-
+        /* we have to wait before we destroy it until it's stopped */
+#else
```

Comment:
Shouldn't we wait for acknowledgement that advertising has ended even for legacy advertising ? 
I understand the code would _work_ for legacy advertising but I would prefer to not expose different paths in the examples.

Reply:
yes, that is another bug, legacy advertising doesn't create the event (yet)

Reply:
anyway, there is no need to destroy the legacy advertising and it stopping or not has no bearing on continuation of the demo - any error will be of course printed